theano/gof/compiledir.py

```diff
@@ -349,35 +349,27 @@ def print_compiledir_content():
     total_key_sizes = 0
     nb_keys = {}
     for dir in os.listdir(compiledir):
-        file = None
-        try:
-            try:
-                filename = os.path.join(compiledir, dir, "key.pkl")
-                file = open(filename, 'rb')
-                keydata = pickle.load(file)
-                ops = list(set([x for x in flatten(keydata.keys)
-                                if isinstance(x, theano.gof.Op)]))
-                if len(ops) == 0:
-                    zeros_op += 1
-                elif len(ops) > 1:
-                    more_than_one_ops += 1
-                else:
-                    types = list(set([x for x in flatten(keydata.keys)
-                                      if isinstance(x, theano.gof.Type)]))
-                    table.append((dir, ops[0], types))
-
-                size = os.path.getsize(filename)
-                total_key_sizes += size
-                if size > max_key_file_size:
-                    big_key_files.append((dir, size, ops))
-
-                nb_keys.setdefault(len(keydata.keys), 0)
-                nb_keys[len(keydata.keys)] += 1
-            except IOError:
-                pass
```

Comment:
Theano very rarely delete key.pkl files. It do this only when they are very old or when the user do this.

So yes, if would be ideal to prevent this race-condition, but in practice, it probably won't happen.

As it wasn't done, can you add the catch to be safe but add a comment to tell why it is like this? That way the next person will know why.